crypto/cryptlib.c

```diff
@@ -101,6 +101,21 @@ void OPENSSL_cpuid_setup(void)
     IA32CAP vec;
     const variant_char *env;
 
+#if defined(__has_feature)
+# if __has_feature(memory_sanitizer)
+    /// This function is called from .init section before memory sanitizer mmaps shadow memory.
+    /// Manually call __msan_init() to initialize it.
+    /// Otherwise, program will crash with segmentation fault when thying access `trigger`,
+    /// because its address was replaced with some not mapped address.
+#if __clang_major__ >= 9
+    __msan_init();
+#else
+    /// Workaround for "error in backend: Sanitizer interface function redefined" in clang-8
+    __asm__ __volatile__("callq __msan_init": : :"memory", "cc", "rax", "rcx", "rdx", "rsi", "rdi", "r8", "r9", "r10", "r11", "st");
```

Comment:
https://www.uclibc.org/docs/psABI-x86_64.pdf page 21
There are some extra registers that may be corrupted by callee